www/src/components/ecosystem/ecosystem-board.js

```diff
@@ -20,48 +24,92 @@ const EcosystemBoardRoot = styled("div")`
   }
 `
 
-const EcosystemBoard = ({
-  icons: { plugins: PluginsIcon, starters: StartersIcon },
-  starters,
-  plugins,
-  background,
-}) => {
-  return (
-    <EcosystemBoardRoot>
-      <EcosystemSection
-        title="Plugins"
-        description="With 429 plugins, lorem ipsum sunt we need some proper copy here pulling in data from your favorite source is only a few mouseclicks away."
-        subTitle="Featured Plugins"
-        icon={PluginsIcon}
-        links={[
-          { label: "Browse Plugins", to: "/plugins/" },
-          {
-            label: "Plugin Authoring",
-            to: "/docs/plugin-authoring/",
-            secondary: true,
-          },
-          { label: "Plugin Docs", to: "/docs/plugins/", secondary: true },
-        ]}
-        featuredItems={plugins}
-      />
-      <EcosystemSection
-        title="Starters"
-        description="Starters are boilerplate Gatsby sites maintained by the community which give you a headstart for your Gatsby project."
-        subTitle="Featured Starters"
-        icon={StartersIcon}
-        links={[
-          { label: "Browse Starters", to: "/starters/" },
-          { label: "Starter Docs", to: "/docs/starters/", secondary: true },
-        ]}
-        featuredItems={starters}
-      />
-      <EcosystemSection
-        title="External Resources"
-        description="More awesome Gatsby content, created by the community."
-        links={[{ label: "Browse Resources", to: "/docs/awesome-gatsby/" }]}
-      />
-    </EcosystemBoardRoot>
-  )
+class EcosystemBoard extends Component {
+  observer
+  observerTargets = []
+
+  componentDidMount() {
+    if (typeof window !== `undefined`) {
+      this.setupObserver()
+    }
+  }
+
+  componentWillUnmount() {
+    for (const target of this.observerTargets) {
+      this.observer.unobserve(target)
+    }
+  }
+
+  setupObserver = () => {
+    const options = { rootMargin: "0px", threshold: [1] }
+    this.observer = new IntersectionObserver(this.handleIntersect, options)
```

Comment:
If we just guard against this not existing, i.e. something like the below, I think this is worth doing!

```jsx
if (typeof window.IntersectionObserver !== 'undefined') {
  // regular setup
} else {
  this.observer = {};
  this.observerTargets = [];
}
```

Reply:
sure, but maybe in componentDidMount
```
  componentDidMount() {
    if (typeof window.IntersectionObserver !== "undefined") {
      this.setupObserver()
    }
  }
```

Reply:
even better; yeah!